runtime/jcl/common/mgmtthread.c

```diff
@@ -765,6 +765,17 @@ Java_com_ibm_java_lang_management_internal_ThreadMXBeanImpl_dumpAllThreadsImpl(J
 	return NULL;
 }
 
+/* Allow access from classes in java.base to use the native method
+ * ThreadMXBeanImpl.dumpAllThreadsImpl() which is in java.management.
```

Comment:
This should, perhaps, be the other way around. The implementation is actually in a library that is part of java.base. The native of `ThreadMXBeanImpl` should call the java.base native.

Reply:
Good idea.  Done.  We could take it further and have  `ThreadMXBeanImpl` call `DiagnosticUtils.dumpAllThreadsImpl()` (which would be made public and exported to `java.management`).

Reply:
I think it's good as it is now.